README.md

```diff
@@ -58,6 +58,33 @@ Installation:
 Easy installation on any distribution supporting snap:  
 https://snapcraft.io/hamster-snap
 
+##### Flatpak
+
+[Flatpak](https://flatpak.org/) enables you to install Hamster in a versioned
+environment and then run it inside a sandbox. It is a method independent from
+your distribution-specific packaging system, ensuring that the application can
+always be reproducibly built, even without hunting down all of the dependencies
+yourself. Debugging is made easier as every user has the exact same environment
+at runtime. Permissions are limited to what the application really needs to
+function properly.
+
+If you downloaded the file with the Hamster bundle (ending in ``.flatpak``), you
+can directly install it with:
+
+```bash
+flatpak install --reinstall Hamster.flatpak
+```
+
+If you would like to install Hamster only for your user, please pass the
+``--user`` option to the above command.
+
+After installation, if you need to invoke Hamster from the command line,
+you can do so with:
+
+```bash
+flatpak run org.gnome.Hamster [args...]
```

Comment:
Would there be any easy way to make this shorter? Maybe flatpack can automically put a symlink in `~/bin` or `/usr/local/bin`? Or otherwise register a short name so you can at least use `flatpack run hamster` (i.e. using the same base command as a normal install)?

Also, it might be good to clarify that (IIUC) the flatpack-hamster will be autostarted through dbus (e.g. when the shell extension tries to access it). That does make me wonder: What does flatpack do when the dbus name is already registered (i.e. when hamster is normally installed as well)? From a quick test it seems to (silently) not claim the name, that is, when I start the flatpack, the flatpack GUI connects to the non-flatpack dbus service, and when I kill the non-flatpack services, it still spawns the non-flatpack ones, rather than the flatpack ones.

For most users, I guess this is not so much an issue (they would typically only install the flatpack version), but it would be at least to document these caveats somewhat. Also, one usecase I see for flatpack is that it would help with testing nightly builds or pullrequests, and then it would be convenient of someone could install the flatpack version next to the regular version and have them completely isolated.

Reply:
> That does make me wonder: What does flatpack do when the dbus name is already registered (i.e. when hamster is normally installed as well)? From a quick test it seems to (silently) not claim the name, that is, when I start the flatpack, the flatpack GUI connects to the non-flatpack dbus service, and when I kill the non-flatpack services, it still spawns the non-flatpack ones, rather than the flatpack ones.

I don't think it's possible to hijack a well-known dbus address. I guess it shouldn't be. At best, hamster itself should log a meaningful message when it fails to acquire the address. But that's independent of packaging. It applies also to a situation with installed hamster and locally (developer) instance, for example.

It's a general problem that I have with all applications that can be installed either "regularly" or as flatpak/snap/appimage (you name it): It's highly intransparent to the user which package she is activating. DBus makes it even more intransparent, of course.

Reply:
> I don't think it's possible to hijack a well-known dbus address. I guess it shouldn't be. At best, hamster itself should log a meaningful message when it fails to acquire the address. But that's independent of packaging. It applies also to a situation with installed hamster and local (developer) instance, for example.

Well, there's two parts to this: One is the claiming of a dbus name at runtime, which is something hamster should report. The other, which I was thinking of, is installing a file in `/usr/share/somewhere` that declares that a dbus interface exists and what binary to autostart if the name is requested but not on the bus yet (at least that's how I gather it works). This is something that `flatpack install` should handle, but maybe flatpack handles this differently? Also, it [seems that the dbus name should match the flatpak name](https://github.com/flatpak/flatpak/issues/1902#issuecomment-405049241), not sure if our `WindowServer` postfix is then supported (the primary service does match).